### ISSUE TITLE
Consul formula

### ISSUE THREAD

**Issue by aphexmunky** [Linuxbrew/homebrew-core#8208]
**Please note we will close your issue without comment if you delete, do not read or do not fill out the issue checklist below and provide ALL the requested information. If you repeatedly fail to use the issue template, we will block you from ever submitting issues to Homebrew again.**

- [X] are reporting a bug others will be able to reproduce and not asking a question. If you're not sure or want to ask a question do so on our Discourse: https://discourse.brew.sh
- [X] have a problem with `brew install` (or `upgrade`, `reinstall`) a single, official formula (not cask)? If it's a general `brew` problem please file this issue at Homebrew/brew: https://github.com/Homebrew/brew/issues/new/choose. If it's a `brew cask` problem please file this issue at https://github.com/Homebrew/homebrew-cask/issues/new/choose. If it's a tap (e.g. Homebrew/homebrew-php) problem please file this issue at the tap.
- [X] ran `brew update` and can still reproduce the problem?
- [X] ran `brew doctor`, fixed all issues and can still reproduce the problem?
- [X] ran `brew gist-logs <formula>` (where `<formula>` is the name of the formula that failed) and included the output link?
- [ ] if `brew gist-logs` didn't work: ran `brew config` and `brew doctor` and included their output with your issue?

To help us debug your issue please explain:
- What you were trying to do (and why)
```
Just upgrading my installs that included consul
```
- What happened (include command output)
```sh
aphex@UK-LT-10831:~$ brew update
Already up-to-date.
aphex@UK-LT-10831:~$ brew upgrade
==> Upgrading 1 outdated package, with result:
consul 1.1.0 -> 1.2.0
==> Upgrading consul 
==> Cloning https://github.com/hashicorp/consul.git
Updating /home/aphex/.cache/Homebrew/consul--git
==> Checking out tag v1.2.0
Error: No such file or directory @ rb_sysopen - scripts/build.sh
```
- What you expected to happen
- Step-by-step reproduction instructions (by running `brew install` commands)

https://gist.github.com/aphexmunky/f2330eba8b132f187bdb566a43db31cd

It looks like they changed their build system in v1.2.0 - I can see they moved the build file to `build-support/scripts/build.sh` and made other changes [here](https://github.com/hashicorp/consul/commit/c90d7f51604f712bd4851a12faae131380b0a9e8#diff-d99b26f864526f6f8f79861d49deb922).

**Comment by iMichka** [comment]
Hi. I saw the error but had no time to look at what changed. Would you be willing to open a pull request to fix this?

**Comment by iMichka** [comment]
Fixed by #8217
